### DISCUSSION TITLE
Vercel CLI local development server does not work with TypeScript API functions

### DISCUSSION THREAD

**Discussion by robvanderleek** [vercel/community#5539] [Bug Report]
### Page to Investigate

https://vercel.com/docs/functions/serverless-functions/quickstart

### Steps to Reproduce

This can be reproduced by following the steps listed in the [page to investigate](https://vercel.com/docs/functions/serverless-functions/quickstart).

For me that results in:
1. `npx create-next-app@latest --typescript`
2. `cd my-app`
3. Create a serverless function in `api/serverless-example.ts`:
```typescript 
import type { VercelRequest, VercelResponse } from '@vercel/node';
 
export default function handler(
  request: VercelRequest,
  response: VercelResponse,
) {
  response.status(200).json({
    body: request.body,
    query: request.query,
    cookies: request.cookies,
  });
}
```
4. `vercel dev`
5. When visiting `http://localhost:3000/api/serverless-example` the output is:
```
Error: Cannot use import statement outside a module
error TS5070: Option '--resolveJsonModule' cannot be specified without 'node' module resolution strategy.

tsconfig.json:10:25 - error TS6046: Argument for '--moduleResolution' option must be: 'node', 'classic', 'node16', 'nodenext'.

10     "moduleResolution": "bundler",
                           ~~~~~~~~~


Found 2 errors in the same file, starting at: tsconfig.json:10
```

Vercel version: 33.2.0
NodeJS version: v20.11.0

**Comment by flesher** [comment]
I was having the exact same issue. and was able to get it working with following changes:

```
// .tsconfig.json
    "module": "Node16",
    "moduleResolution": "node16",

// package.json
   "type": "module"
```

**Comment by robvanderleek** [comment]
Hi @flesher 

Thanks for your reply.

Updating the configuration files with the settings you suggest leads to a lot of
```
Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'
```
errors.

Eventually, I switched my project to NextJS ([this is the project](https://github.com/robvanderleek/create-issue-branch) in case you're interested).

It seems `next dev` gives a lot less headaches than `vercel dev`. I really tried to get things going with `vercel dev` since my project is not a frontend application (doesn't use React), but failed. Note that everything works on the Vercel platform but running TypeScript serverless functions locally with `vercel dev` seems really hard/impossible.
